exporter/exporter.go

```diff
@@ -11,38 +10,65 @@ import (
 
 //go:generate counterfeiter -o mocks/cfclient.go . CFClient
 type CFClient interface {
-	ListAppsByQuery(url.Values) ([]cfclient.App, error)
+	ListAppsWithSpaceAndOrg() ([]cfclient.App, error)
 }
 
+// Stuct to store all names related to an app (app name, space name, org name) so we can track if these have changed
+// for a given app and if so delete and recreate it's app watcher
+type cfNames struct {
+	appName   string
+	spaceName string
+	orgName   string
+}
+
+func newCfNames(appName string, spaceName string, orgName string) cfNames {
+	return cfNames{
+		appName:   appName,
+		spaceName: spaceName,
+		orgName:   orgName,
+	}
+}
+
+// TODO guid type
+
 type PaasExporter struct {
 	cf             CFClient
 	watcherManager WatcherManager
-	appNameByGuid  map[string]string
+	cfNamesByGuid  map[string]cfNames
 }
 
-func New(cf CFClient, wm WatcherManager) *PaasExporter {
+func NewPaasExporter(cf CFClient, wm WatcherManager) *PaasExporter {
```

Comment:
@henrytk (hi!) suggested that `New` is idiomatic go when you're creating an instance of a type that is the core thing in the package.  ie, when calling this in main.go, we say `exporter.New(...)` and it's clear that we're creating an exporter from the package name.